Move to Fedora 31!

jlebon opened the pull request:
See individual commit messages.

Requires: https://github.com/coreos/rpm-ostree/pull/1927

Marking as draft until the f30 release is completely done (and I'd like to investigate the `efivar-libs` bit a little more).

<details>

<summary>Diff from latest `testing` release</summary>

```
Upgraded:
  NetworkManager 1:1.16.4-1.fc30 -> 1:1.20.4-1.fc31
  NetworkManager-libnm 1:1.16.4-1.fc30 -> 1:1.20.4-1.fc31
  acl 2.2.53-3.fc30 -> 2.2.53-4.fc31
  adcli 0.8.2-3.fc30 -> 0.8.2-7.fc31
  afterburn 4.1.3-1.module_f30+6561+89082b01 -> 4.1.3-1.module_f31+6563+49f1a537
  afterburn-dracut 4.1.3-1.module_f30+6561+89082b01 -> 4.1.3-1.module_f31+6563+49f1a537
  alternatives 1.11-4.fc30 -> 1.11-5.fc31
  audit-libs 3.0-0.9.20190507gitf58ec40.fc30 -> 3.0-0.12.20190507gitf58ec40.fc31
  avahi-libs 0.7-18.fc30 -> 0.7-20.fc31
  basesystem 11-7.fc30 -> 11-8.fc31
  bash 5.0.7-1.fc30 -> 5.0.7-3.fc31
  bash-completion 1:2.8-6.fc30 -> 1:2.8-7.fc31
  bc 1.07.1-8.fc30 -> 1.07.1-9.fc31
  bind-libs 32:9.11.10-1.fc30 -> 32:9.11.11-1.fc31
  bind-libs-lite 32:9.11.10-1.fc30 -> 32:9.11.11-1.fc31
  bind-license 32:9.11.10-1.fc30 -> 32:9.11.11-1.fc31
  bind-utils 32:9.11.10-1.fc30 -> 32:9.11.11-1.fc31
  brotli 1.0.7-3.fc30 -> 1.0.7-6.fc31
  btrfs-progs 5.2.1-1.fc30 -> 5.2.1-1.fc31
  bubblewrap 0.3.3-2.fc30 -> 0.3.3-3.fc31
  bzip2 1.0.6-29.fc30 -> 1.0.8-1.fc31
  bzip2-libs 1.0.6-29.fc30 -> 1.0.8-1.fc31
  c-ares 1.15.0-4.module_f30+5262+10ab6e6f -> 1.15.0-4.module_f31+5271+b9c46c4d
  ca-certificates 2018.2.26-3.fc30 -> 2019.2.32-3.fc31
  chrony 3.5-1.fc30 -> 3.5-4.fc31
  cifs-utils 6.9-1.fc30 -> 6.9-2.fc31
  cloud-utils-growpart 0.31-2.fc30 -> 0.31-3.fc31
  console-login-helper-messages 0.17-1.fc30 -> 0.17-1.fc31
  console-login-helper-messages-issuegen 0.17-1.fc30 -> 0.17-1.fc31
  console-login-helper-messages-motdgen 0.17-1.fc30 -> 0.17-1.fc31
  console-login-helper-messages-profile 0.17-1.fc30 -> 0.17-1.fc31
  container-selinux 2:2.117.0-1.gitbfde70a.fc30 -> 2:2.117.0-1.gitbfde70a.fc31
  containernetworking-plugins 0.8.2-2.git485be65.fc30 -> 0.8.2-2.1.dev.git485be65.fc31
  containers-common 1:0.1.37-2.fc30 -> 1:0.1.39-2.2.dev.git1cf1e06.fc31
  coreos-installer 0-9.git2fcf441.fc30 -> 0-10.git2fcf441.fc31
  coreos-installer-dracut 0-9.git2fcf441.fc30 -> 0-10.git2fcf441.fc31
  coreutils 8.31-2.fc30 -> 8.31-4.fc31
  coreutils-common 8.31-2.fc30 -> 8.31-4.fc31
  cpio 2.12-10.fc30 -> 2.12-12.fc31
  cracklib 2.9.6-19.fc30 -> 2.9.6-21.fc31
  crypto-policies 20190527-1.git0b3add8.fc30 -> 20190816-4.gitbb9bf99.fc31
  cryptsetup 2.2.1-1.fc30 -> 2.2.1-1.fc31
  cryptsetup-libs 2.2.1-1.fc30 -> 2.2.1-1.fc31
  cups-libs 1:2.2.12-2.fc30 -> 1:2.2.12-2.fc31
  curl 7.65.3-4.fc30 -> 7.66.0-1.fc31
  cyrus-sasl-gssapi 2.1.27-0.6rc7.fc30 -> 2.1.27-2.fc31
  cyrus-sasl-lib 2.1.27-0.6rc7.fc30 -> 2.1.27-2.fc31
  dbus 1:1.12.16-1.fc30 -> 1:1.12.16-3.fc31
  dbus-broker 21-3.fc30 -> 21-6.fc31
  dbus-common 1:1.12.16-1.fc30 -> 1:1.12.16-3.fc31
  dbus-libs 1:1.12.16-1.fc30 -> 1:1.12.16-3.fc31
  dbxtool 8-9.fc30 -> 8-10.fc31
  device-mapper 1.02.154-3.fc30 -> 1.02.163-2.fc31
  device-mapper-event 1.02.154-3.fc30 -> 1.02.163-2.fc31
  device-mapper-event-libs 1.02.154-3.fc30 -> 1.02.163-2.fc31
  device-mapper-libs 1.02.154-3.fc30 -> 1.02.163-2.fc31
  device-mapper-multipath 0.7.9-6.git2df6110.fc30 -> 0.8.0-3.fc31
  device-mapper-multipath-libs 0.7.9-6.git2df6110.fc30 -> 0.8.0-3.fc31
  device-mapper-persistent-data 0.7.6-4.fc30 -> 0.8.5-2.fc31
  dhcp-client 12:4.3.6-37.fc30 -> 12:4.4.1-15.fc31
  dhcp-common 12:4.3.6-37.fc30 -> 12:4.4.1-15.fc31
  diffutils 3.7-2.fc30 -> 3.7-3.fc31
  dosfstools 4.1-8.fc30 -> 4.1-9.fc31
  dracut 049-26.git20181204.fc30 -> 049-27.git20181204.fc31.1
  dracut-network 049-26.git20181204.fc30 -> 049-27.git20181204.fc31.1
  e2fsprogs 1.44.6-1.fc30 -> 1.45.3-1.fc31
  e2fsprogs-libs 1.44.6-1.fc30 -> 1.45.3-1.fc31
  efi-filesystem 4-2.fc30 -> 4-3.fc31
  efibootmgr 16-5.fc30 -> 16-6.fc31
  elfutils-default-yama-scope 0.177-1.fc30 -> 0.177-1.fc31
  elfutils-libelf 0.177-1.fc30 -> 0.177-1.fc31
  elfutils-libs 0.177-1.fc30 -> 0.177-1.fc31
  expat 2.2.8-1.fc30 -> 2.2.8-1.fc31
  fedora-coreos-pinger 0.0.4-1.module_f30+5099+4493ad29 -> 0.0.4-1.module_f31+5371+3c747891
  fedora-gpg-keys 30-2 -> 31-1
  fedora-release-common 30-5 -> 31-1
  fedora-release-coreos 30-5 -> 31-1
  fedora-repos 30-2 -> 31-1
  fedora-repos-ostree 30-2 -> 31-1
  file 5.36-4.fc30 -> 5.37-3.fc31
  file-libs 5.36-4.fc30 -> 5.37-3.fc31
  filesystem 3.10-1.fc30 -> 3.12-2.fc31
  findutils 1:4.6.0-22.fc30 -> 1:4.6.0-24.fc31
  fipscheck 1.5.0-6.fc30 -> 1.5.0-7.fc31
  fipscheck-lib 1.5.0-6.fc30 -> 1.5.0-7.fc31
  freetype 2.9.1-7.fc30 -> 2.10.0-3.fc31
  fstrm 0.4.0-3.fc30 -> 0.4.0-4.fc31
  fuse 2.9.9-7.fc30 -> 2.9.9-8.fc31
  fuse-common 3.6.2-1.fc30 -> 3.6.2-1.fc31
  fuse-libs 2.9.9-7.fc30 -> 2.9.9-8.fc31
  fuse-overlayfs 0.6.3-2.0.dev.git46c0f8e.fc30 -> 0.6.4-2.fc31
  fuse-sshfs 3.5.2-1.fc30 -> 3.5.2-2.fc31
  fuse3 3.6.2-1.fc30 -> 3.6.2-1.fc31
  fuse3-libs 3.6.2-1.fc30 -> 3.6.2-1.fc31
  gawk 4.2.1-6.fc30 -> 5.0.1-5.fc31
  gdbm-libs 1:1.18-4.fc30 -> 1:1.18.1-1.fc31
  gdisk 1.0.4-4.fc30 -> 1.0.4-5.fc31
  gettext 0.19.8.1-22.fc30 -> 0.20.1-2.fc31
  gettext-libs 0.19.8.1-22.fc30 -> 0.20.1-2.fc31
  glib2 2.60.7-1.fc30 -> 2.62.1-1.fc31
  glibc 2.29-22.fc30 -> 2.30-5.fc31
  glibc-all-langpacks 2.29-22.fc30 -> 2.30-5.fc31
  glibc-common 2.29-22.fc30 -> 2.30-5.fc31
  gmp 1:6.1.2-10.fc30 -> 1:6.1.2-10.fc31
  gnupg2 2.2.17-1.fc30 -> 2.2.17-2.fc31
  gnutls 3.6.8-1.fc30 -> 3.6.10-1.fc31
  gpgme 1.12.0-1.fc30 -> 1.13.1-3.fc31
  grep 3.1-9.fc30 -> 3.3-3.fc31
  grub2-common 1:2.02-81.fc30 -> 1:2.02-100.fc31
  grub2-efi-x64 1:2.02-81.fc30 -> 1:2.02-100.fc31
  grub2-pc 1:2.02-81.fc30 -> 1:2.02-100.fc31
  grub2-pc-modules 1:2.02-81.fc30 -> 1:2.02-100.fc31
  grub2-tools 1:2.02-81.fc30 -> 1:2.02-100.fc31
  grub2-tools-extra 1:2.02-81.fc30 -> 1:2.02-100.fc31
  grub2-tools-minimal 1:2.02-81.fc30 -> 1:2.02-100.fc31
  gzip 1.9-9.fc30 -> 1.10-1.fc31
  hostname 3.20-8.fc30 -> 3.20-9.fc31
  ipcalc 0.2.5-2.fc30 -> 0.2.5-3.fc31
  iproute 5.2.0-1.fc30 -> 5.3.0-1.fc31
  iproute-tc 5.2.0-1.fc30 -> 5.3.0-1.fc31
  iptables 1.8.2-3.fc30 -> 1.8.3-5.fc31
  iptables-libs 1.8.2-3.fc30 -> 1.8.3-5.fc31
  iptables-nft 1.8.2-3.fc30 -> 1.8.3-5.fc31
  iptables-services 1.8.2-3.fc30 -> 1.8.3-5.fc31
  iputils 20180629-4.fc30 -> 20190515-3.fc31
  irqbalance 2:1.4.0-3.fc30 -> 2:1.4.0-4.fc31
  iscsi-initiator-utils 6.2.0.876-8.gitf3c8e90.fc30 -> 6.2.0.876-10.gitf3c8e90.fc31
  iscsi-initiator-utils-iscsiuio 6.2.0.876-8.gitf3c8e90.fc30 -> 6.2.0.876-10.gitf3c8e90.fc31
  isns-utils-libs 0.97-8.fc30 -> 0.97-9.fc31
  jansson 2.12-2.fc30 -> 2.12-4.fc31
  jq 1.6-2.fc30 -> 1.6-3.fc31
  json-c 0.13.1-4.fc30 -> 0.13.1-6.fc31
  json-glib 1.4.4-2.fc30 -> 1.4.4-3.fc31
  kbd 2.0.4-13.fc30 -> 2.0.4-14.fc31
  kbd-legacy 2.0.4-13.fc30 -> 2.0.4-14.fc31
  kbd-misc 2.0.4-13.fc30 -> 2.0.4-14.fc31
  kernel 5.2.18-200.fc30 -> 5.3.4-300.fc31
  kernel-core 5.2.18-200.fc30 -> 5.3.4-300.fc31
  kernel-modules 5.2.18-200.fc30 -> 5.3.4-300.fc31
  keyutils 1.6-2.fc30 -> 1.6-3.fc31
  keyutils-libs 1.6-2.fc30 -> 1.6-3.fc31
  kmod 26-3.fc30 -> 26-4.fc31
  kmod-libs 26-3.fc30 -> 26-4.fc31
  kpartx 0.7.9-6.git2df6110.fc30 -> 0.8.0-3.fc31
  krb5-libs 1.17-15.fc30 -> 1.17-45.fc31
  less 551-1.fc30 -> 551-2.fc31
  libacl 2.2.53-3.fc30 -> 2.2.53-4.fc31
  libaio 0.3.111-4.fc30 -> 0.3.111-6.fc31
  libarchive 3.3.3-6.fc30 -> 3.4.0-1.fc31
  libargon2 20161029-8.fc30 -> 20171227-3.fc31
  libassuan 2.5.2-2.fc30 -> 2.5.3-2.fc31
  libattr 2.4.48-5.fc30 -> 2.4.48-7.fc31
  libbasicobjects 0.1.1-42.fc30 -> 0.1.1-43.fc31
  libblkid 2.33.2-2.fc30 -> 2.34-3.fc31
  libcap 2.26-5.fc30 -> 2.26-6.fc31
  libcap-ng 0.7.9-7.fc30 -> 0.7.9-8.fc31
  libcollection 0.7.0-42.fc30 -> 0.7.0-43.fc31
  libcom_err 1.44.6-1.fc30 -> 1.45.3-1.fc31
  libcroco 0.6.13-1.fc30 -> 0.6.13-2.fc31
  libcurl 7.65.3-4.fc30 -> 7.66.0-1.fc31
  libdb 5.3.28-37.fc30 -> 5.3.28-38.fc31
  libdb-utils 5.3.28-37.fc30 -> 5.3.28-38.fc31
  libdhash 0.5.0-42.fc30 -> 0.5.0-43.fc31
  libedit 3.1-27.20190324cvs.fc30 -> 3.1-28.20190324cvs.fc31
  libevent 2.1.8-7.fc30 -> 2.1.8-7.fc31
  libfdisk 2.33.2-2.fc30 -> 2.34-3.fc31
  libffi 3.1-19.fc30 -> 3.1-23.fc31
  libgcc 9.2.1-1.fc30 -> 9.2.1-1.fc31
  libgcrypt 1.8.5-1.fc30 -> 1.8.5-1.fc31
  libgomp 9.2.1-1.fc30 -> 9.2.1-1.fc31
  libgpg-error 1.33-2.fc30 -> 1.36-2.fc31
  libidn2 2.2.0-1.fc30 -> 2.2.0-2.fc31
  libini_config 1.3.1-42.fc30 -> 1.3.1-43.fc31
  libipa_hbac 2.2.2-1.fc30 -> 2.2.2-1.fc31
  libkcapi 1.1.4-1.fc30 -> 1.1.5-1.fc31
  libkcapi-hmaccalc 1.1.4-1.fc30 -> 1.1.5-1.fc31
  libksba 1.3.5-9.fc30 -> 1.3.5-10.fc31
  libldb 1.5.5-1.fc30 -> 2.0.7-1.fc31
  libmetalink 0.1.3-8.fc30 -> 0.1.3-9.fc31
  libmnl 1.0.4-9.fc30 -> 1.0.4-10.fc31
  libmodulemd1 1.8.15-3.fc30.1 -> 1.8.15-3.fc31
  libmount 2.33.2-2.fc30 -> 2.34-3.fc31
  libndp 1.7-3.fc30 -> 1.7-4.fc31
  libnetfilter_conntrack 1.0.7-2.fc30 -> 1.0.7-3.fc31
  libnfnetlink 1.0.1-15.fc30 -> 1.0.1-16.fc31
  libnfsidmap 1:2.4.1-0.rc1.fc30 -> 1:2.4.1-1.rc1.fc31
  libnftnl 1.1.3-1.fc30 -> 1.1.3-2.fc31
  libnghttp2 1.39.2-1.module_f30+5957+05aeb4f9 -> 1.39.2-1.module_f31+5955+08aea14a
  libnl3 3.4.0-8.fc30 -> 3.5.0-1.fc31
  libnsl2 1.2.0-4.20180605git4a062cf.fc30 -> 1.2.0-5.20180605git4a062cf.fc31
  libpath_utils 0.2.1-42.fc30 -> 0.2.1-43.fc31
  libpcap 14:1.9.0-3.fc30 -> 14:1.9.0-4.fc31
  libpkgconf 1.6.1-1.fc30 -> 1.6.3-2.fc31
  libpng 2:1.6.36-1.fc30 -> 2:1.6.37-2.fc31
  libpsl 0.20.2-6.fc30 -> 0.21.0-2.fc31
  libpwquality 1.4.0-12.fc30 -> 1.4.1-1.fc31
  libref_array 0.1.5-42.fc30 -> 0.1.5-43.fc31
  libreport-filesystem 2.10.1-1.fc30 -> 2.10.1-2.fc31
  libseccomp 2.4.1-0.fc30 -> 2.4.1-1.fc31
  libselinux 2.9-3.1.fc30 -> 2.9-5.fc31
  libselinux-utils 2.9-3.1.fc30 -> 2.9-5.fc31
  libsemanage 2.9-1.fc30 -> 2.9-3.fc31
  libsepol 2.9-1.fc30 -> 2.9-2.fc31
  libsigsegv 2.11-7.fc30 -> 2.11-8.fc31
  libsmartcols 2.33.2-2.fc30 -> 2.34-3.fc31
  libsmbclient 2:4.10.8-0.fc30 -> 2:4.11.0-3.fc31
  libsolv 0.7.5-1.fc30 -> 0.7.5-3.fc31
  libss 1.44.6-1.fc30 -> 1.45.3-1.fc31
  libssh 0.9.0-5.fc30 -> 0.9.0-6.fc31
  libssh-config 0.9.0-5.fc30 -> 0.9.0-6.fc31
  libsss_certmap 2.2.2-1.fc30 -> 2.2.2-1.fc31
  libsss_idmap 2.2.2-1.fc30 -> 2.2.2-1.fc31
  libsss_nss_idmap 2.2.2-1.fc30 -> 2.2.2-1.fc31
  libstdc++ 9.2.1-1.fc30 -> 9.2.1-1.fc31
  libtalloc 2.1.16-1.fc30 -> 2.3.0-1.fc31
  libtasn1 4.13-7.fc30 -> 4.14-2.fc31
  libtdb 1.3.18-1.fc30 -> 1.4.2-1.fc31
  libtevent 0.9.39-1.fc30 -> 0.10.1-1.fc31
  libtirpc 1.1.4-2.rc3.fc30 -> 1.1.4-2.rc3.fc31
  libunistring 0.9.10-5.fc30 -> 0.9.10-6.fc31
  libusbx 1.0.22-2.fc30 -> 1.0.22-4.fc31
  libuser 0.62-20.fc30 -> 0.62-21.fc31
  libutempter 1.1.6-16.fc30 -> 1.1.6-17.fc31
  libuuid 2.33.2-2.fc30 -> 2.34-3.fc31
  libvarlink-util 18-1.fc30 -> 18-2.fc31
  libverto 0.3.0-7.fc30 -> 0.3.0-8.fc31
  libwbclient 2:4.10.8-0.fc30 -> 2:4.11.0-3.fc31
  libxcrypt 4.4.10-1.fc30 -> 4.4.10-1.fc31
  libxml2 2.9.9-2.fc30 -> 2.9.9-3.fc31
  libyaml 0.2.1-5.fc30 -> 0.2.2-2.fc31
  libzstd 1.4.2-1.fc30 -> 1.4.2-1.fc31
  linux-atm-libs 2.5.1-25.fc30 -> 2.5.1-25.fc31
  linux-firmware 20190923-102.fc30 -> 20190923-102.fc31
  linux-firmware-whence 20190923-102.fc30 -> 20190923-102.fc31
  lmdb-libs 0.9.23-2.fc30 -> 0.9.23-3.fc31
  logrotate 3.15.1-1.fc30 -> 3.15.1-1.fc31
  lua-libs 5.3.5-5.fc30 -> 5.3.5-6.fc31
  lvm2 2.02.183-3.fc30 -> 2.03.05-2.fc31
  lvm2-libs 2.02.183-3.fc30 -> 2.03.05-2.fc31
  lz4-libs 1.9.1-1.fc30 -> 1.9.1-1.fc31
  lzo 2.08-15.fc30 -> 2.08-16.fc31
  mdadm 4.1-rc2.0.3.fc30 -> 4.1-rc2.0.5.fc31.1
  microcode_ctl 2:2.1-32.fc30 -> 2:2.1-32.fc31
  moby-engine 18.06.3-2.ce.gitd7080c1.fc30 -> 18.09.8-2.ce.git0dd43dd.fc31
  mozjs60 60.9.0-2.fc30 -> 60.9.0-3.fc31
  mpfr 3.1.6-4.fc30 -> 3.1.6-5.fc31
  ncurses 6.1-10.20180923.fc30 -> 6.1-12.20190803.fc31
  ncurses-base 6.1-10.20180923.fc30 -> 6.1-12.20190803.fc31
  ncurses-libs 6.1-10.20180923.fc30 -> 6.1-12.20190803.fc31
  net-tools 2.0-0.54.20160912git.fc30 -> 2.0-0.55.20160912git.fc31
  nettle 3.4.1rc1-2.fc30 -> 3.5.1-3.fc31
  nfs-utils-coreos 1:2.4.1-0.rc1.fc30 -> 1:2.4.1-1.rc1.fc31
  nftables 1:0.9.1-2.fc30 -> 1:0.9.1-3.fc31
  nmap-ncat 2:7.70-7.fc30 -> 2:7.80-2.fc31
  npth 1.6-2.fc30 -> 1.6-3.fc31
  nss-altfiles 2.18.1-14.fc30 -> 2.18.1-15.fc31
  numactl-libs 2.0.12-2.fc30 -> 2.0.12-3.fc31
  oniguruma 6.9.2-2.fc30 -> 6.9.3-1.fc31
  openldap 2.4.47-1.fc30 -> 2.4.47-3.fc31
  openssh 8.0p1-5.fc30 -> 8.0p1-8.fc31.1
  openssh-clients 8.0p1-5.fc30 -> 8.0p1-8.fc31.1
  openssh-server 8.0p1-5.fc30 -> 8.0p1-8.fc31.1
  openssl 1:1.1.1d-2.fc30 -> 1:1.1.1d-2.fc31
  openssl-libs 1:1.1.1d-2.fc30 -> 1:1.1.1d-2.fc31
  os-prober 1.74-8.fc30 -> 1.77-3.fc31
  ostree 2019.4-2.fc30 -> 2019.4-3.fc31
  ostree-libs 2019.4-2.fc30 -> 2019.4-3.fc31
  p11-kit 0.23.16.1-1.fc30 -> 0.23.16.1-2.fc31
  p11-kit-trust 0.23.16.1-1.fc30 -> 0.23.16.1-2.fc31
  pam 1.3.1-17.fc30 -> 1.3.1-18.fc31
  passwd 0.80-5.fc30 -> 0.80-6.fc31
  pcre 8.43-2.fc30 -> 8.43-2.fc31.1
  pcre2 10.33-14.fc30 -> 10.33-14.fc31
  pigz 2.4-4.fc30 -> 2.4-5.fc31
  pkgconf 1.6.1-1.fc30 -> 1.6.3-2.fc31
  pkgconf-m4 1.6.1-1.fc30 -> 1.6.3-2.fc31
  pkgconf-pkg-config 1.6.1-1.fc30 -> 1.6.3-2.fc31
  podman 2:1.5.1-3.fc30 -> 2:1.5.1-3.git0005792.fc31
  policycoreutils 2.9-4.fc30 -> 2.9-5.fc31
  polkit 0.116-2.fc30 -> 0.116-4.fc31
  polkit-libs 0.116-2.fc30 -> 0.116-4.fc31
  polkit-pkla-compat 0.1-14.fc30 -> 0.1-15.fc31
  popt 1.16-17.fc30 -> 1.16-18.fc31
  procps-ng 3.3.15-5.fc30 -> 3.3.15-6.fc31
  protobuf-c 1.3.1-2.fc30 -> 1.3.1-3.fc31
  publicsuffix-list-dafsa 20190128-2.fc30 -> 20190417-2.fc31
  python-pip-wheel 19.0.3-3.fc30 -> 19.1.1-4.fc31
  python-setuptools-wheel 40.8.0-1.fc30 -> 41.2.0-1.fc31
  python3-libs 3.7.4-1.fc30 -> 3.7.4-5.fc31
  qrencode-libs 3.4.4-8.fc30 -> 4.0.2-4.fc31
  readline 8.0-2.fc30 -> 8.0-3.fc31
  rpm 4.14.2.1-5.fc30 -> 4.15.0-1.fc31
  rpm-libs 4.14.2.1-5.fc30 -> 4.15.0-1.fc31
  rpm-ostree 2019.6-1.fc30 -> 2019.6-1.fc31
  rpm-ostree-libs 2019.6-1.fc30 -> 2019.6-1.fc31
  rpm-plugin-selinux 4.14.2.1-5.fc30 -> 4.15.0-1.fc31
  rsync 3.1.3-8.fc30 -> 3.1.3-9.fc31
  runc 2:1.0.0-93.dev.gitb9b6cc6.fc30 -> 2:1.0.0-101.rc9.gitc1485a1.fc31
  samba-client-libs 2:4.10.8-0.fc30 -> 2:4.11.0-3.fc31
  samba-common 2:4.10.8-0.fc30 -> 2:4.11.0-3.fc31
  samba-common-libs 2:4.10.8-0.fc30 -> 2:4.11.0-3.fc31
  samba-libs 2:4.10.8-0.fc30 -> 2:4.11.0-3.fc31
  sed 4.5-3.fc30 -> 4.5-4.fc31
  selinux-policy 3.14.3-46.fc30 -> 3.14.4-37.fc31
  selinux-policy-targeted 3.14.3-46.fc30 -> 3.14.4-37.fc31
  setup 2.13.3-1.fc30 -> 2.13.3-2.fc31
  sg3_utils 1.42-7.fc30 -> 1.42-8.fc31
  sg3_utils-libs 1.42-7.fc30 -> 1.42-8.fc31
  shadow-utils 2:4.6-8.fc30 -> 2:4.6-16.fc31
  skopeo 1:0.1.37-2.fc30 -> 1:0.1.39-2.2.dev.git1cf1e06.fc31
  slirp4netns 0.4.0-4.git19d199a.fc30 -> 0.4.0-20.1.dev.gitbbd6f25.fc31
  sqlite-libs 3.26.0-6.fc30 -> 3.29.0-2.fc31
  sssd 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-ad 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-client 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-common 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-common-pac 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-ipa 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-krb5 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-krb5-common 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sssd-ldap 2.2.2-1.fc30 -> 2.2.2-1.fc31
  sudo 1.8.27-1.fc30 -> 1.8.27-3.fc31
  systemd 241-12.git323cdf4.fc30 -> 243-2.gitfab6f01.fc31
  systemd-container 241-12.git323cdf4.fc30 -> 243-2.gitfab6f01.fc31
  systemd-libs 241-12.git323cdf4.fc30 -> 243-2.gitfab6f01.fc31
  systemd-pam 241-12.git323cdf4.fc30 -> 243-2.gitfab6f01.fc31
  systemd-rpm-macros 241-12.git323cdf4.fc30 -> 243-2.gitfab6f01.fc31
  systemd-udev 241-12.git323cdf4.fc30 -> 243-2.gitfab6f01.fc31
  tar 2:1.32-1.fc30 -> 2:1.32-2.fc31
  tzdata 2019c-1.fc30 -> 2019c-1.fc31
  userspace-rcu 0.10.1-5.fc30 -> 0.11.1-2.fc31
  util-linux 2.33.2-2.fc30 -> 2.34-3.fc31
  vim-minimal 2:8.1.1991-2.fc30 -> 2:8.1.2102-1.fc31
  which 2.21-14.fc30 -> 2.21-15.fc31
  whois-nls 5.5.2-1.fc30 -> 5.5.2-1.fc31
  xfsprogs 4.19.0-5.fc30 -> 5.1.0-2.fc31
  xz 5.2.4-5.fc30 -> 5.2.4-6.fc31
  xz-libs 5.2.4-5.fc30 -> 5.2.4-6.fc31
  zchunk-libs 1.1.2-2.fc30 -> 1.1.2-3.fc31
  zincati 0.0.5-1.module_f30+6699+1eaa40c6 -> 0.0.5-1.module_f31+6700+1b2863ea
  zlib 1.2.11-18.fc30 -> 1.2.11-19.fc31
Removed:
  GeoIP-1.6.12-5.fc30.x86_64
  GeoIP-GeoLite-data-2018.06-3.fc30.noarch
  bind-export-libs-32:9.11.10-1.fc30.x86_64
  dhcp-libs-12:4.3.6-37.fc30.x86_64
  libtool-ltdl-2.4.6-29.fc30.x86_64
  timedatex-0.6-1.fc30.x86_64
Added:
  compat-readline5-5.2-34.fc31.x86_64
  conmon-2:2.0.1-1.fc31.x86_64
  containerd-1.2.6-2.20190627gitd68b593.fc31.x86_64
  crun-0.10.2-1.fc31.x86_64
  libmaxminddb-1.2.0-8.fc31.x86_64
  libtextstyle-0.20.1-2.fc31.x86_64
  python3-3.7.4-5.fc31.x86_64
  yajl-2.1.0-13.fc31.x86_64
```

</details>

dustymabe commented:
> > 
> 
> My initial reason for not doing it for this transition is that I didn't want to gate moving to f31 on getting `next`/`next-devel` set up. The earlier we move to f31 the better.

+1 

> We have that with the lockfile diff, though I can print the rpm-ostree generated pkgdiff as a comment too.

Yeah I see that let's analyze the changes to see if there are any red flags.

jlebon commented:
Added diff to first comment. And here's the interesting bit duplicated:

```
Removed:
  GeoIP-1.6.12-5.fc30.x86_64
  GeoIP-GeoLite-data-2018.06-3.fc30.noarch
  bind-export-libs-32:9.11.10-1.fc30.x86_64
  dhcp-libs-12:4.3.6-37.fc30.x86_64
  libtool-ltdl-2.4.6-29.fc30.x86_64
  timedatex-0.6-1.fc30.x86_64
Added:
  compat-readline5-5.2-34.fc31.x86_64
  conmon-2:2.0.1-1.fc31.x86_64
  containerd-1.2.6-2.20190627gitd68b593.fc31.x86_64
  crun-0.10.2-1.fc31.x86_64
  libmaxminddb-1.2.0-8.fc31.x86_64
  libtextstyle-0.20.1-2.fc31.x86_64
  python3-3.7.4-5.fc31.x86_64
  yajl-2.1.0-13.fc31.x86_64
```

`python3` is from https://github.com/coreos/fedora-coreos-tracker/issues/280.
`containerd` is from `moby-engine`.
`conmon` and `crun` are from `podman`.

dustymabe commented:
for the others:

```
# rpm -e compat-readline5 libmaxminddb libtextstyle yajl
error: Failed dependencies:
        libreadline.so.5()(64bit) is needed by (installed) lvm2-2.03.05-2.fc31.x86_64
        libmaxminddb.so.0()(64bit) is needed by (installed) bind-libs-lite-32:9.11.11-1.fc31.x86_64
        libmaxminddb.so.0()(64bit) is needed by (installed) bind-libs-32:9.11.11-1.fc31.x86_64
        libtextstyle.so.0()(64bit) is needed by (installed) gettext-libs-0.20.1-2.fc31.x86_64
        libtextstyle.so.0()(64bit) is needed by (installed) gettext-0.20.1-2.fc31.x86_64
        libyajl.so.2()(64bit) is needed by (installed) crun-0.10.2-1.fc31.x86_64
```

- `compat-readline5` is now needed by `lvm`
- `bind-libs` grew a dep for `libmaxminddb`
- `gettext` grew a dep for `libtextstyle`
- `crun` needs `yajl`

dustymabe commented:
we can merge this after https://github.com/coreos/coreos-assembler/pull/854 correct?

jlebon commented:
OK, this one is ready to go! :rocket:

dustymabe commented:
kola failures in CI

```
    --- FAIL: podman.base (33.97s)
    --- PASS: podman.base/info (0.79s)
    --- FAIL: podman.base/resources (4.58s)
            cluster.go:122: STEP 1: FROM scratch
            cluster.go:122: STEP 2: COPY . /
            cluster.go:122: STEP 3: COMMIT localhost/echo
            cluster.go:122: Getting image source signatures
            cluster.go:122: Copying blob sha256:b1f6a31602870db4d7c75f59c9e788883c0c2f14658c7f8d748f5c41d4f8dd41
            cluster.go:122: Copying config sha256:83e186be137d822f6eacb7e7547ce25abcdaac6e959e7849957edcd4d216645f
            cluster.go:122: Writing manifest to image destination
            cluster.go:122: Storing signatures
            cluster.go:122: Error: cannot set kernel memory with cgroupv2: File exists: OCI runtime error
            podman.go:279: Failed to run "sudo podman run --net=none --rm --kernel-memory=10m echo echo 1": output: "" status: "Process exited with status 127"
```


`cannot set kernel memory with cgroupv2: File exists: OCI runtime error` is the interesting part

jlebon commented:
https://github.com/coreos/mantle/issues/1103

jlebon commented:
Right, so I think this is https://github.com/coreos/fedora-coreos-tracker/issues/292. I think it was mostly decided that we'd ship with v1 at this point. So we need to tweak some knobs to default to that. And that should hopefully fix the podman tests.

If it turns out we *are* shipping v2 in stable, we can always revert.

jlebon commented:
And the last one is:

```
--- FAIL: coreos.ignition.resource.remote (2548.52s)
        harness.go:486: Cluster failed starting machines: machine "45f32065-a58b-404f-909f-865582167026" failed to start: ssh journalctl failed: ssh: handshake failed: read tcp 127.0.0.1:47564->127.0.0.1:42889: read: connection reset by peer
```

Still looking into that one.

arithx commented:
@jlebon: that's likely a flake, I'd recommend re-running that test specifically.

jlebon commented:
OK, that should be all the failures!

cgwalters commented:
One messy thing with this is that we don't have a way to cleanly distinguish between kargs *we* set by default, and ones users set.  This is https://github.com/ostreedev/ostree/issues/479

So in the future, we'll be by default staying with the non-unified hierarchy unless we do something like write a special systemd unit that changes the kargs.

Not a blocker, just noting this.

dustymabe commented:
still LGTM

jlebon commented:
New podman failures, fun!

jlebon commented:
OK so somehow adding back `network-legacy` flips network availability around. We have networking in the initrd, but not in the real root. Something like NetworkManager expecting to be used in both or not at all?

jlebon commented:
OK, I think I got to the bottom of this finally. Just going to copy/paste from the relevant commit message:


    fedora-coreos-base: drop nm-initrd-generator

    In Fedora 31, NetworkManager includes an initrd module which replaces
    the legacy dracut networking one. The key bit is here:

    https://github.com/dracutdevs/dracut/blob/1fcc70fe57eea0ea658aa2de5c0044683fe85cf1/modules.d/40network/module-setup.sh#L11.

    It was initially in f30 as well, but was reverted due to issues:
    https://src.fedoraproject.org/rpms/NetworkManager/c/7d3054a9e366f1ebdd9d7854bf11a3069119d6da?branch=f30

    (One thing that confused me at first: note that this isn't a systemd
    generator, it's run via the dracut cmdline hook.)

    The generator does seem to have feature parity wrt karg support:
    https://github.com/NetworkManager/NetworkManager/blob/ba64c162dc471740a405f9d59e7a132f0e290078/src/initrd/nmi-cmdline-reader.c

    Though I couldn't get it to work in my tests; `rd.neednet=1` does not
    result in a populated `resolv.conf`, which would of course cause
    Ignition to fail.

    Anyway, given that we tie pretty strongly into initrd networking for
    e.g. Ignition and the installer, I think I'd rather we stick with the
    status quo for now, and do the transition separately from the FCOS f31
    rebase.

jlebon commented:
> :heavy_check_mark: **continuous-integration/jenkins/pr-merge** — This commit looks good 

Woohoo!
